javascript/javascript2/week3/hyf-homework-week3/exercise.js

```diff
@@ -0,0 +1,173 @@
+console.log('Js 2 homework week 3')
+       
+// step 1
+setTimeout(() => {
+    console.log("Called after 2.5 seconds")
+}, 2500);
+
+    // OR
+// window.addEventListener('load',alreadyLoaded)
+// function alreadyLoaded() {
+//     setTimeout(() => {
+//         console.log("Called after 2.5 seconds")
+//     }, 2500);  
+// }
+
+//         // step 2
+
+function loadString(delay,stringToLog) {
+    setTimeout(() => {
+        console.log(stringToLog)
+    }, delay);
+}
+loadString(5000,"Called after 5 seconds");
+
+//         // step 3
+
+const firstButton = document.querySelector('#btn1')
+firstButton.addEventListener('click',() =>{
+    loadString(5000,"Button clicked after 5 seconds");
+})
+
+        // step 
+
+const forEarth = (() => {
+    console.log('Earth')
+})
+// forEarth()
+
+const forSaturn= (() => {
+    console.log('Saturn')
+})
+// forSaturn()
+
+ function planetLogFunction(callBack){
+    return callBack()
+}
+planetLogFunction(forEarth)
+function planetLogFunction(callBack){
+    return callBack()
+}
+planetLogFunction(forSaturn)
+
+//                 // step 5
+
+const logButton = document.querySelector('#btn2')
+logButton.addEventListener('click',getLocation)
+const buttonText = document.getElementById("demo");
+function getLocation() {
+    if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(showPosition);
+    } else {
+        buttonText.innerHTML = "Geolocation is not supported by this browser.";
+    }
+}
+function showPosition(position) {
+    buttonText.innerHTML = "Latitude: " + position.coords.latitude + 
+    "<br>Longitude: " + position.coords.longitude; 
+}
+//                 // step 6 Optiional
+let buttonMap = document.getElementById("get-location");
+let locationlink = document.querySelector('#locationlink');
+buttonMap.addEventListener("click", function() {
+  navigator.geolocation.getCurrentPosition(function(position) {
+    let lat = position.coords.latitude;
+    let long = position.coords.longitude;
+    locationlink.href= `https://maps.google.com/maps?q=${lat},${long}&hl=es;z=14&amp;output=embed`;
+    locationlink.innerText= ` Your location is ${lat}° Lattitude by ${ long}° Longitude`
+  });
+});
+
+
+//                 // step 7
+function runAfterDelay(delay,callback){
+    setTimeout(() => {
+        return callback()
+    }, delay);
+}
+runAfterDelay(3000,forEarth)
+
+function runAfterDelay(delay,callback){
+    setTimeout(() => {
+        return callback()
+    }, delay);
+}
+runAfterDelay(10000,forSaturn)
+
```

Comment:
same as above. only one time you need to define the function

Reply:
done